src/include/walfile.h

```diff
@@ -30,9 +30,224 @@
 #define PGMONETA_WALFILE_H
 
 #include <deque.h>
-#include <walfile/wal_reader.h>
 
+#include <stdbool.h>
```

Comment:
You need to describe why we need all of this in `walfile.h`...

None of it are used in the functions...

Reply:
Yeah I remember my ide was complaining about not `bool` and some `int` not being defined so I added these, but now with the file changes it seems to be okay so I removed them.

Reply:
That is ok, but I meant all the changes in `walfile.h` - we only need the display function for now.

Remember that `walfile.h` should be as small as possible

Reply:
I see. The reason for needing almost all of them is because we have this function`char* pgmoneta_wal_encode_xlog_record` in `walfile.c` used by writing functionality. 
But now I'm thinking about trying and bringing this function to `wal_reader` to reduce complexity in `walfile`.

Reply:
ok I've been able to revert almost all of additions to `walfile.h` to make it as simple as possible, now it only has:

```
struct xlog_long_page_header_data;
struct walfile;
int pgmoneta_read_walfile(int server, char* path, struct walfile** wf);
int pgmoneta_write_walfile(struct walfile* wf, int server, char* path);
void pgmoneta_destroy_walfile(struct walfile* wf);
int pgmoneta_describe_walfile(char* path, enum value_type type);
```
I haven't pushed this yet

Reply:
That sounds great !

Reply:
Something like resource manager and record type will come into walfile.h later (2 steps from now - well, 3 including this one)